pkg/pod/status.go

```diff
@@ -255,7 +253,7 @@ func createMessageFromResults(results []v1beta1.PipelineResourceResult) (string,
 	return string(bytes), nil
 }
 
-func filterResultsAndResources(results []v1beta1.PipelineResourceResult, specResults []v1beta1.TaskResult) ([]v1beta1.TaskRunResult, []v1beta1.PipelineResourceResult, []v1beta1.PipelineResourceResult) {
+func filterResultsAndResources(results []v1beta1.PipelineResourceResult, specResults []v1beta1.TaskResult) ([]v1beta1.TaskRunResult, []v1beta1.PipelineResourceResult) {
```

Comment:
can you add a docstring for this function? I'm confused why pipelineresourceresults are still used here-- can we update the code that uses them?

Reply:
A good docstring should explain all of the inputs and all of the outputs, for example:

```
filterResultsAndResources filters the input results based on the results declared in the task spec.
It returns a slice of any of the input results that are defined in the task spec, converted to TaskRunResults,
and a slice of any of the input results that don't represent internal values (i.e. those that should not be displayed in the TaskRun status.
```